site/en/blog/private-network-access-preflight/index.md

```diff
@@ -0,0 +1,334 @@
+Access-Control-Allow-Private-Network: true
+```
+
+Alternatively, the server may set `Access-Control-Allow-Origin: *`, though this
+is dangerous and discouraged. Private network resources should rarely be
+accessible to all origins, so think carefully about the risks involved in
+setting such a header.
+
+Then Chrome will send the actual request:
+
+```text
+HTTP/1.1 GET /cat.gif
+...
+```
+
+To which the server can respond normally.
+
+
+#### CORS mode
+
+Say `https://foo.example/index.html` runs the following code:
+
+```text
+await fetch('https://bar.example/delete-everything', {
+  method: 'PUT',
+  credentials: 'include',
+})
+```
+
+Again, say `bar.example`  resolves to `192.168.1.1`.
+
+Chrome first sends a preflight request:
+
+```text
+HTTP/1.1 OPTIONS /give-me-root
+Origin: https://foo.example
+Access-Control-Request-Method: PUT
+Access-Control-Request-Credentials: true
+Access-Control-Request-Private-Network: true
+```
+
+For this request to succeed, the server must respond with:
+
+```text
+HTTP/1.1 204 No Content
+Access-Control-Allow-Origin: https://foo.example
+Access-Control-Allow-Methods: PUT
+Access-Control-Allow-Credentials: true
+Access-Control-Allow-Private-Network: true
+```
+
+Then Chrome will send the actual request:
+
+```text
+HTTP/1.1 PUT /give-me-root
+Origin: https://foo.example
+```
+
+To which the server can respond per usual CORS rules:
+
+```text
+HTTP/1.1 200 OK
+Access-Control-Allow-Origin: https://foo.example
+```
+
+## How to know if your website is affected
+
+Starting in Chrome 98, if a private network request is detected, a preflight
+request will be sent ahead of it. If this preflight request fails, the final
+request will still be sent, but a warning will be surfaced in the DevTools
+issues panel.
+
+{% Img
+   src="image/VbsHyyQopiec0718rMq2kTE1hke2/AgZzPf3NkMWQ0Cm6Puu0.png",
+   alt="A failed preflight request warning in the Devtools Issues panel",
+   class="screenshot",
+   width="800", height="556"
+%}
+
+Affected preflight requests can also be viewed and diagnosed in the network panel:
+
+{% Img
+   src="image/VbsHyyQopiec0718rMq2kTE1hke2/aysOX5wKA1kme8HyV3t0.png",
+   alt="A failed preflight request in the DevTools Network panel",
+   class="screenshot",
+   width="800", height="265"
+%}
+
+To see what happens if preflight success was enforced, you can
+[pass the following command-line argument](https://www.chromium.org/developers/how-tos/run-chromium-with-flags)
+starting in Chrome 98:
+
+```text
+–enable-features=PrivateNetworkAccessRespectPreflightResutls
+```
+
+Any failed preflight request will result in a failed fetch. This can allow you
+to test whether your website would work after the
+[second phase of our rollout plan](#rollout-plan). Errors can be diagnosed in
+the same way as warnings using the DevTools panels mentioned above.
+
+## What to do if your website is affected
+
+While this change is not expected to break any website, we strongly encourage
```

Comment:
This message assumes the readers see this immediately after this article is published, but the future readers may find this confusing. Can we be more timeline agnostic?